### DISCUSSION TITLE
en/latest/examples/workflows/upstream/cluster_workflow_template__workflow_template_ref/

### DISCUSSION THREAD

**Discussion by giscus** [argoproj-labs/hera#1089] [Website Discussions]
# en/latest/examples/workflows/upstream/cluster_workflow_template__workflow_template_ref/

https://hera.readthedocs.io/en/latest/examples/workflows/upstream/cluster_workflow_template__workflow_template_ref/?h=template+ref

**Comment by jiantang99** [comment]
Hi, do you have an example where multiple template refs are used in 1 workflow? I.e., first step will run the first template, and the second step will run the second template.

**Reply by elliotgunton** [reply]
Hey, this example is for "workflow template refs" - you are looking for regular "template refs" used in Steps/Dags. Here's an example for what you describe: https://hera.readthedocs.io/en/stable/examples/workflows/upstream/workflow_template__dag/ Thanks for the question!
